api/src/test/java/org/openmrs/api/VisitServiceTest.java

```diff
@@ -963,32 +965,74 @@ public void saveVisit_shouldAssociateEncounterWithVisitOnSaveEncounter() throws
 	 * @see VisitService#saveVisit(Visit)
 	 */
 	@Test
-	@Verifies(value = "save visit should persist new encounter", method = "saveVisit(Visit)")
-	public void saveVisit_shouldPersistNewEncounter() throws Exception {
+	@Verifies(value = "save visit should not persist new encounter", method = "saveVisit(Visit)")
+	public void saveVisit_shouldNotPersistNewEncounter() throws Exception {
 		
 		VisitService vs = Context.getVisitService();
 		Visit visit = vs.getVisit(1);
 		
 		Encounter encounter = new Encounter();
 		encounter.setEncounterDatetime(new Date());
+		encounter.setEncounterType(Context.getEncounterService().getEncounterType(1));
 		encounter.setPatient(visit.getPatient());
 		encounter.setLocation(visit.getLocation());
 		visit.addEncounter(encounter);
 		
 		vs.saveVisit(visit);
-		
+
+		assertNull(encounter.getEncounterId()); // should not have saved the encounter
+
+		Context.getEncounterService().saveEncounter(encounter); //save the encounter
+		Integer encounterId = encounter.getEncounterId();
+
 		Context.flushSession();
 		Context.clearSession();
-		
-		Integer encounterId = encounter.getEncounterId();
-		
+
 		// reload the visit
 		visit = Context.getVisitService().getVisit(1);
-		
+
 		assertEquals(1, visit.getEncounters().size());
 		assertEquals(encounterId, ((Encounter) visit.getEncounters().toArray()[0]).getEncounterId());
 	}
-	
+
+	/**
+	 * @see VisitService#saveVisit(Visit)
+	 */
+	@Test
+	@Verifies(value = "save visit should not update existing encounter", method = "saveVisit(Visit)")
+	public void saveVisit_shouldNotUpdateExistingEncounter() throws Exception {
+		executeDataSet(VISITS_WITH_DATES_XML);
+
+		VisitService vs = Context.getVisitService();
+		Visit visit = vs.getVisit(7);
+		Encounter encounter = visit.getNonVoidedEncounters().get(0);
+		assertEquals(0, encounter.getAllObs(true).size());
+		int obsBeforeCount = encounter.getAllObs(true).size();
+		Obs obs = new Obs();
+		obs.setConcept(Context.getConceptService().getConcept(19));
+		obs.setValueText("new obs");
+		encounter.addObs(obs);
+
+		vs.saveVisit(visit);
+
+		assertNull(encounter.getObs().iterator().next().getObsId()); // should not have saved the encounter
```

Comment:
When you delete a parent with cascading disabled, my assumption is that hibernate will delete the parent and just drop the references to it from the oprhaned children meaning the children stick around standing on their own and we need a test to prove this behavior. Choosing to tell hibernate to delete vs keep 'orphan' children is purely based on real world semantics i.e can an encounter exist without a visit? Based on our data model this is true where as for certain objects like ConceptName that can't exist on their own based on our data model we explicitly tell hibernate to delete them when they become orphans.

As for saving a parent with a new child object, disabling the cascade would result in a hibernate exception complaining about a reference to a transient associated object so yes it might be worth adding a test that ensures this is happening.